python/dgl/transforms/module.py

```diff
@@ -501,6 +548,13 @@ def __call__(self, g):
                 data_dict[c_etype] = g.edges(etype=c_etype)
 
             g = update_graph_structure(g, data_dict)
+
+            for c_etype in g.canonical_etypes:
+                if c_etype[0] == c_etype[2] and c_etype[1] != 'self':
+                    fill_data = {k: v[-g.num_nodes(c_etype[0]):, :] for k, v in
+                                 g.edges[c_etype].data.items()}
+                    g.edges[(c_etype[0], 'self', c_etype[2])].data.update(fill_data)
+
```

Comment:
I don't think L552-556 are needed. These are new edge types and they do not have features.